rbi/annotations/activesupport.rbi

```diff
@@ -142,8 +142,11 @@ class Array
   sig { params(object: T.untyped).returns(T::Array[T.untyped]) }
   def self.wrap(object); end
 
-  sig { returns(T.untyped) }
-  def extract!; end
+  sig do
+    params(block: T.nilable(T.proc.params(element: Elem).returns(T.untyped)))
+      .returns(T::Array[T::Array[Elem]])
```

Comment:
I think adding the block is fine. I'm not so sure about this return though. I don't think this will work for invocations of `extract!` without a block, which [return an enumerator](https://github.com/rails/rails/blob/7d1c4dd7be1b51e2dcbb3614b68dc98c2926d28c/activesupport/lib/active_support/core_ext/array/extract.rb#L11).

Also, why is the return a nested array rather a simple one?